core/services/workflows/engine.go

```diff
@@ -687,6 +689,7 @@ func (e *Engine) finishExecution(ctx context.Context, executionID string, status
 	e.stepUpdatesChMap.remove(executionID)
 	metrics.updateTotalWorkflowsGauge(ctx, e.stepUpdatesChMap.len())
 	metrics.updateWorkflowExecutionLatencyGauge(ctx, executionDuration)
+	metrics.incrementWorkflowExecutionFinishedCounter(ctx)
```

Comment:
Could we add a status tag to this?

Reply:
It already has a status tag, at least the status of the status we receive on the param. `metrics` here is `e.metrics.with("status", status)`

Reply:
Awesome!